pkg/resource/deploy/builtins.go

```diff
@@ -142,20 +142,29 @@ func (p *builtinProvider) Read(urn resource.URN, id resource.ID,
 	}, resource.StatusOK, nil
 }
 
+const readStackOutputs = "pulumi:pulumi:readStackOutputs"
 const readStackResourceOutputs = "pulumi:pulumi:readStackResourceOutputs"
 
 func (p *builtinProvider) Invoke(tok tokens.ModuleMember,
 	args resource.PropertyMap) (resource.PropertyMap, []plugin.CheckFailure, error) {
-	if tok != readStackResourceOutputs {
-		return nil, nil, errors.Errorf("unrecognized function name: '%v'", tok)
-	}
 
-	outs, err := p.readStackResourceOutputs(args)
-	if err != nil {
-		return nil, nil, err
-	}
+	if tok == readStackOutputs {
+		outs, err := p.readStackReference(args)
```

Comment:
reusing readStackReference (what we call in "readResource" in the constructor of StackReference) as it produces the exact values we want.